server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -150,4 +188,44 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
 
     }
+
+    private void validate(
+        ClusterState currentState,
+        Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap,
+        boolean restoreAllShards
+    ) {
+        String errorMsg = "cannot restore index [%s] because an open index with same name already exists in the cluster.";
+        for (Map.Entry<String, Tuple<Boolean, IndexMetadata>> indexMetadataEntry : indexMetadataMap.entrySet()) {
+            String indexName = indexMetadataEntry.getKey();
+            IndexMetadata indexMetadata = indexMetadataEntry.getValue().v2();
+            String indexUuid = indexMetadata.getIndexUUID();
+            boolean fromRemoteStore = indexMetadataEntry.getValue().v1();
+            if (indexMetadata.getSettings().getAsBoolean(SETTING_REMOTE_STORE_ENABLED, false)) {
+                if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState()) == false) {
+                    throw new IllegalStateException(String.format(Locale.ROOT, errorMsg, indexName) + " Close the existing index.");
+                }
+
+                if (fromRemoteStore) {
+                    boolean sameNameIndexExists = currentState.metadata().hasIndex(indexName);
```

Comment:
If index is deleted but present in index graveyard, would there be any conflicts ?

Reply:
As remote state is up-to-date, there shouldnt be a case where same index is present in remote but deleted in local. But good to assert as this is an assumption

Reply:
These cases need to be handled for Point-in-time restores.

Reply:
yeah, for PIT restore we would need to handle them. But for now i'll add an assert for this